docs/examples/robot_keyboard_2.py

```diff
@@ -12,9 +12,9 @@
 must_have = {i for i in range(1, 32)}
 must_not_have = {0}
 devices = [
-    device
-    for device in devices
-    for keys in (set(device.capabilities().get(ecodes.EV_KEY, [])),)
+    dev
+    for dev in devices
+    for keys in (set(dev.capabilities().get(ecodes.EV_KEY, [])),)
```

Comment:
Heh, I was more referring to the multiple usages of `devices` and nested list comprehensions - if I was writing this myself I might have done something like:
```keyboard_devices = filter(is_keyboard, evdev_devices)```
IYSWIM  \*shrug\*